src/main/java/com/diffplug/gradle/spotless/FormatTask.java

```diff
@@ -23,19 +23,36 @@
 import java.util.List;
 import java.util.Locale;
 
+import com.diffplug.common.base.Errors;
 import org.gradle.api.DefaultTask;
 import org.gradle.api.GradleException;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputFiles;
+import org.gradle.api.tasks.OutputFiles;
+import org.gradle.api.tasks.SkipWhenEmpty;
 import org.gradle.api.tasks.TaskAction;
+import org.gradle.api.tasks.incremental.IncrementalTaskInputs;
 
 public class FormatTask extends DefaultTask {
 	// set by SpotlessExtension, but possibly overridden by FormatExtension
+	@Input
 	public Charset encoding = StandardCharsets.UTF_8;
-	public LineEnding.Policy lineEndingsPolicy = LineEnding.UNIX_POLICY;
+	@Input
+	public LineEnding.Policy lineEndingPolicy = LineEnding.UNIX_POLICY;
+
 	// set by FormatExtension
+	@Input
 	public boolean paddedCell = false;
+	@InputFiles
+	@OutputFiles
+	@SkipWhenEmpty
 	public Iterable<File> target;
+	@Input
```

Comment:
I'm pretty sure nobody would use a `spotlessFastCheck` if you have to run `spotlessCheck` to be sure it's correct anyway.

The only real alternative I see is properly supporting up-to-date checks. Instead of referencing `Throwing.Function` directly, we should reference a serializable sublcass of it. Maybe call it `FormatterFunction`. Then implement the default functions (like regex) in such a way that they properly support equals/hashcode too. 

The worst thing that could happen after that is that the task is out-of-date unnecessarily when users use inline steps. The readme about "Making spotless faster" could then explain that users should write real classes with equals/hashcode instead to make the task fast.